src/components/autocompleter/autocompleter.jsx

```diff
@@ -0,0 +1,105 @@
+import PropTypes from 'prop-types';
+import React, { forwardRef, useEffect, useState } from 'react';
+import useFetch from '@bloodyaugust/use-fetch';
+import Select from '../select/select.jsx';
+import ListOption from '../list-option/list-option.jsx';
+import mockConstants from '../../mocks/mock-constants.js';
+
+const { API_ROOT } = mockConstants;
+
+const Autocompleter = forwardRef(function Autocompleter(props, ref) {
+  const { execute: fetch } = useFetch();
+  const [models, setModels] = useState(props.models);
+
+  useEffect(fetchModels, []);
+
+  function fetchModels(searchString) {
+    fetch(apiEndpoint(searchString))
+      .then((results) => {
+        setModels(results.json.results.map(resultInfo => results.json[resultInfo.key][resultInfo.id]));
+      }).catch((error) => {
+        if (error.error && error.error.type !== 'aborted') {
+          throw error;
+        }
+      });
+  }
+
+  function apiEndpoint(searchString) {
+    if (searchString) {
+      return `${API_ROOT}${props.apiEndpoint}?${props.searchParam}=${searchString}`;
+    }
+
+    return `${API_ROOT}${props.apiEndpoint}`;
+  }
+
+  function onKeyUp(event) {
```

Comment:
We might want to consider using `onInput` instead.

> Note: If you're looking for a way to react to changes in an input's value, you should use the input event. Some changes are not detectable by keyup, for example pasting text from the context menu in a text input.

-- https://developer.mozilla.org/en-US/docs/Web/API/Document/keyup_event

We don't want to search when a user is just pressing modifier keys, as an example.